src/Datadog.Trace.Tools.Runner/Datadog.Trace.Tools.Runner.proj

```diff
@@ -0,0 +1,167 @@
+      <Output TaskParameter="TargetOutputs" ItemName="CollectedBuildOutput"/>
+    </MSBuild>
+
+  </Target>
+
+  <Target Name="BuildNativeWindows">
+    <Message Importance="high" Text="%0A *** Native projects to build: %0A   @(NativeWinProject, '%0A   ')%0A" />
+
+    <Message Importance="high" Text="%0A *** Building Native ClrProfiler for Win-x64 *** %0A" />
+    <MSBuild Targets="Build" Projects="@(NativeWinProject)" Properties="Configuration=Release;Platform=x64" />
+
+    <Message Importance="high" Text="%0A *** Building Native ClrProfiler for Win-x86 *** %0A" />
+    <MSBuild Targets="Build" Projects="@(NativeWinProject)" Properties="Configuration=Release;Platform=x86" />
+
+    <ItemGroup>
+      <NativeWinX64Output Include="..\*Native\bin\Release\x64\*.dll" />
+      <NativeWinX86Output Include="..\*Native\bin\Release\x86\*.dll" />
+    </ItemGroup>
+
+    <Copy SourceFiles="@(NativeWinX64Output)" DestinationFolder="@(NativeWinX64OutputFolder)"
+          SkipUnchangedFiles="true" Retries="3" RetryDelayMilliseconds="300" />
+
+    <Copy SourceFiles="@(NativeWinX86Output)" DestinationFolder="@(NativeWinX86OutputFolder)"
+          SkipUnchangedFiles="true" Retries="3" RetryDelayMilliseconds="300" />
+  </Target>
+
+  <Target Name="BuildNativeLinux" DependsOnTargets="BuildNativeLinux_Exist;BuildNativeLinux_NotExist">
+  </Target>
+
+  <Target Name="BuildNativeLinux_Exist" Condition="Exists(@(NativeLinuxX64Output))">
+    <Message Importance="high" Text="A linux binary already exist in the folder, skipping the build." />
+  </Target>
+
+  <Target Name="BuildNativeLinux_NotExist" Condition="!Exists(@(NativeLinuxX64Output))">
+    <Message Importance="high" Text="%0A *** Checking for docker daemon...%0A" />
+    <CheckAnyProcessExist ProcessNames="com.docker.proxy;dockerd">
+      <Output TaskParameter="ReturnValue" PropertyName="IsDockerRunning" />
+    </CheckAnyProcessExist>
+    <Error Text="Docker is not running. Docker is required to build the Linux native binaries." Condition="!$(IsDockerRunning)" />
+
+    <Exec Command="linux-native-build.cmd" />
+  </Target>
+
+  <Target Name="Clean">
+    <MSBuild Targets="Clean" Projects="Datadog.Trace.Tools.Runner.csproj" Properties="Configuration=Release;" />
+    <ItemGroup>
+      <NativeWinX64Output Include="..\*Native\bin\Release\x64\*.dll" />
+      <NativeWinX86Output Include="..\*Native\bin\Release\x86\*.dll" />
+    </ItemGroup>
+    <Delete Files="@(NativeLinuxX64Output)" />
+    <Delete Files="@(NativeWinX64Output)" />
+    <Delete Files="@(NativeWinX86Output)" />
+  </Target>
+
+  <UsingTask TaskName="CheckAnyProcessExist" TaskFactory="RoslynCodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
+    <ParameterGroup>
+      <ProcessNames ParameterType="System.String" Required="true" />
+      <ReturnValue ParameterType="System.Boolean" Output="true" />
+    </ParameterGroup>
+    <Task>
```

Comment:
🤯 I didn't realize you could author managed Tasks inside an MSBuild file

Reply:
haha this opens a new world of possibilities...